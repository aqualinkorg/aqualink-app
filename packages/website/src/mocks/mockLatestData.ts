import { LatestData } from 'store/Sites/types';

const aDayAgo = new Date(
  new Date().setDate(new Date().getDate() - 1),
).toISOString();

export const mockLatestData: LatestData[] = [
  {
    id: 8597,
    timestamp: aDayAgo,
    value: 0,
    source: 'noaa',
    metric: 'temp_alert',
    siteId: 1,
  },
  {
    id: 8660,
    timestamp: aDayAgo,
    value: 0,
    source: 'noaa',
    metric: 'temp_weekly_alert',
    siteId: 1,
  },
  {
    id: 8599,
    timestamp: aDayAgo,
    value: 0,
    source: 'noaa',
    metric: 'dhw',
    siteId: 1,
  },
  {
    id: 8600,
    timestamp: aDayAgo,
    value: 15.3299999237061,
    source: 'noaa',
    metric: 'satellite_temperature',
    siteId: 1,
  },
  {
    id: 3534,
    timestamp: aDayAgo,
    value: 20.160000000000004,
    source: 'spotter',
    metric: 'top_temperature',
    siteId: 1,
  },
  {
    id: 3508,
    timestamp: aDayAgo,
    value: 20.259999999999998,
    source: 'spotter',
    metric: 'bottom_temperature',
    siteId: 1,
  },
  {
    id: 63633,
    timestamp: aDayAgo,
    value: 30.177,
    source: 'sonde',
    metric: 'bottom_temperature',
    siteId: 1,
  },
  {
    id: 8598,
    timestamp: aDayAgo,
    value: -0.4800000762939014,
    source: 'noaa',
    metric: 'sst_anomaly',
    siteId: 1,
  },
  {
    id: 3450,
    timestamp: aDayAgo,
    value: 0.63,
    source: 'spotter',
    metric: 'significant_wave_height',
    siteId: 1,
  },
  {
    id: 3595,
    timestamp: aDayAgo,
    value: 5.9,
    source: 'spotter',
    metric: 'wave_mean_period',
    siteId: 1,
  },
  {
    id: 3433,
    timestamp: aDayAgo,
    value: 152.354,
    source: 'spotter',
    metric: 'wave_mean_direction',
    siteId: 1,
  },
  {
    id: 3601,
    timestamp: aDayAgo,
    value: 1.2,
    source: 'spotter',
    metric: 'wind_speed',
    siteId: 1,
  },
  {
    id: 3551,
    timestamp: aDayAgo,
    value: 262,
    source: 'spotter',
    metric: 'wind_direction',
    siteId: 1,
  },
  {
    id: 63608,
    timestamp: aDayAgo,
    value: 0.08,
    source: 'sonde',
    metric: 'cholorophyll_rfu',
    siteId: 1,
  },
  {
    id: 63610,
    timestamp: aDayAgo,
    value: 0.35,
    source: 'sonde',
    metric: 'cholorophyll_concentration',
    siteId: 1,
  },
  {
    id: 63611,
    timestamp: aDayAgo,
    value: 60847.4,
    source: 'sonde',
    metric: 'conductivity',
    siteId: 1,
  },
  {
    id: 63613,
    timestamp: aDayAgo,
    value: 7.922,
    source: 'sonde',
    metric: 'water_depth',
    siteId: 1,
  },
  {
    id: 63615,
    timestamp: aDayAgo,
    value: 111.9,
    source: 'sonde',
    metric: 'odo_saturation',
    siteId: 1,
  },
  {
    id: 63617,
    timestamp: aDayAgo,
    value: 6.9,
    source: 'sonde',
    metric: 'odo_concentration',
    siteId: 1,
  },
  {
    id: 63619,
    timestamp: aDayAgo,
    value: 36.56,
    source: 'sonde',
    metric: 'salinity',
    siteId: 1,
  },
  {
    id: 63620,
    timestamp: aDayAgo,
    value: 55372.3,
    source: 'sonde',
    metric: 'specific_conductance',
    siteId: 1,
  },
  {
    id: 63622,
    timestamp: aDayAgo,
    value: 35992,
    source: 'sonde',
    metric: 'tds',
    siteId: 1,
  },
  {
    id: 63624,
    timestamp: aDayAgo,
    value: 2.44,
    source: 'sonde',
    metric: 'turbidity',
    siteId: 1,
  },
  {
    id: 63626,
    timestamp: aDayAgo,
    value: 0,
    source: 'sonde',
    metric: 'total_suspended_solids',
    siteId: 1,
  },
  {
    id: 63627,
    timestamp: aDayAgo,
    value: 1.187,
    source: 'sonde',
    metric: 'sonde_wiper_position',
    siteId: 1,
  },
  {
    id: 63629,
    timestamp: aDayAgo,
    value: 8.26,
    source: 'sonde',
    metric: 'ph',
    siteId: 1,
  },
  {
    id: 63631,
    timestamp: aDayAgo,
    value: -80.7,
    source: 'sonde',
    metric: 'ph_mv',
    siteId: 1,
  },
  {
    id: 63635,
    timestamp: aDayAgo,
    value: 5.97,
    source: 'sonde',
    metric: 'sonde_battery_voltage',
    siteId: 1,
  },
  {
    id: 63636,
    timestamp: aDayAgo,
    value: 0,
    source: 'sonde',
    metric: 'sonde_cable_power_voltage',
    siteId: 1,
  },
];
