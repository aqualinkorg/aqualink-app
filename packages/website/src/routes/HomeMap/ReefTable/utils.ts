import { startCase } from "lodash";
import type { TableRow } from "../../../store/Homepage/types";

export type Order = "asc" | "desc";

export enum OrderKeys {
  ALERT = "alert",
  DEPTH = "depth",
  DHW = "dhw",
  LOCATION_NAME = "locationName",
  TEMP = "temp",
}

// This type isn't used anywhere, it just forces the above enum to only hold valid keys to TableRow.
type __CheckOrderKeys = TableRow[OrderKeys];

/**
 * Returns a human friendly string of a order key.
 * @param orderKey key to return the string for.
 */
export function getOrderKeysFriendlyString(orderKey: OrderKeys) {
  switch (orderKey) {
    case OrderKeys.DHW:
      return "DHW";
    default:
      return startCase(orderKey);
  }
}

export function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
  return b[orderBy] >= a[orderBy] ? 1 : -1;
}

export function getComparator(
  order: Order,
  orderBy: OrderKeys
): (
  a: {
    [key in OrderKeys]: number | string | null;
  },
  b: {
    [key in OrderKeys]: number | string | null;
  }
) => number {
  return order === "desc"
    ? (a, b) => descendingComparator(a, b, orderBy)
    : (a, b) => -descendingComparator(a, b, orderBy);
}

export function stableSort<T>(array: T[], comparator: (a: T, b: T) => number) {
  // eslint-disable-next-line fp/no-mutating-methods
  return [...array].sort(comparator);
}
