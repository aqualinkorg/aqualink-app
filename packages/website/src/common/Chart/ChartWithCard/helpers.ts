import { minBy, maxBy, meanBy } from "lodash";
import moment from "moment";
import {
  findMarginalDate,
  generateHistoricalMonthlyMeanTimestamps,
} from "../../../helpers/dates";

import {
  DailyData,
  HistoricalMonthlyMean,
  HistoricalMonthlyMeanData,
  SofarValue,
  TimeSeriesData,
} from "../../../store/Reefs/types";
import { filterHistoricalMonthlyMeanData } from "../utils";
import { CardColumn } from "./types";

export const calculateCardMetrics = (
  minNumberOfPoints: number,
  data?: (HistoricalMonthlyMeanData | SofarValue)[],
  keyPrefix?: string
): CardColumn["rows"] => [
  {
    key: `${keyPrefix}-max`,
    value: data?.[minNumberOfPoints - 1]
      ? maxBy(data, "value")?.value
      : undefined,
  },
  {
    key: `${keyPrefix}-mean`,
    value: data?.[minNumberOfPoints - 1] ? meanBy(data, "value") : undefined,
  },
  {
    key: `${keyPrefix}-min`,
    value: data?.[minNumberOfPoints - 1]
      ? minBy(data, "value")?.value
      : undefined,
  },
];

// Show at least 3 ticks on the chart
export const findChartPeriod = (startDate: string, endDate: string) => {
  const from = moment(new Date(startDate).toISOString());
  const to = moment(new Date(endDate).toISOString());
  const week = 7;
  const month = 30;
  const diffDays = to.diff(from, "days");

  switch (true) {
    case diffDays < 2:
      return "hour";
    case diffDays < 3 * week:
      return "day";
    case diffDays < 3 * month:
      return "week";
    default:
      return "month";
  }
};

export const findDataLimits = (
  historicalMonthlyMean: HistoricalMonthlyMean[],
  dailyData: DailyData[] | undefined,
  timeSeriesData: TimeSeriesData | undefined,
  startDate: string | undefined,
  endDate: string | undefined
): [string | undefined, string | undefined] => {
  const { hobo, spotter } = timeSeriesData || {};
  const historicalMonthlyMeanData = generateHistoricalMonthlyMeanTimestamps(
    historicalMonthlyMean,
    startDate,
    endDate
  );
  const filteredHistoricalMonthlyMeanData = filterHistoricalMonthlyMeanData(
    historicalMonthlyMeanData,
    startDate,
    endDate
  );

  const hasData = Boolean(
    filteredHistoricalMonthlyMeanData?.[0] ||
      dailyData?.[0] ||
      spotter?.bottomTemperature?.[0] ||
      spotter?.topTemperature?.[0] ||
      hobo?.bottomTemperature?.[0]
  );

  return [
    hasData
      ? new Date(
          findMarginalDate(
            filteredHistoricalMonthlyMeanData,
            dailyData || [],
            spotter,
            hobo?.bottomTemperature || [],
            "min"
          )
        ).toISOString()
      : undefined,
    hasData
      ? new Date(
          findMarginalDate(
            filteredHistoricalMonthlyMeanData,
            dailyData || [],
            spotter,
            hobo?.bottomTemperature || []
          )
        ).toISOString()
      : undefined,
  ];
};

/**
 * Determines if the input dates differ by more than one year
 * @param startDate - The starting date
 * @param endDate - The ending date
 */
export const moreThanOneYear = (startDate: string, endDate: string) => {
  const from = moment(startDate);
  const to = moment(endDate);
  return to.diff(from, "years") >= 1;
};
