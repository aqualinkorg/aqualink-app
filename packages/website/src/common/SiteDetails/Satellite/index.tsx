import React from "react";
import {
  withStyles,
  WithStyles,
  createStyles,
  Card,
  CardContent,
  Typography,
  CardHeader,
  Grid,
  Tooltip,
  Box,
} from "@material-ui/core";

import { dhwColorCode } from "../../../assets/colorCode";
import type { LiveData } from "../../../store/Reefs/types";
import { formatNumber } from "../../../helpers/numberUtils";
import satellite from "../../../assets/satellite.svg";
import {
  dhwColorFinder,
  degreeHeatingWeeksCalculator,
} from "../../../helpers/degreeHeatingWeeks";
import { styles as incomingStyles } from "../styles";
import UpdateInfo from "../../UpdateInfo";
import { toRelativeTime } from "../../../helpers/dates";

const Satellite = ({ maxMonthlyMean, liveData, classes }: SatelliteProps) => {
  const { degreeHeatingDays, satelliteTemperature, sstAnomaly } = liveData;
  const relativeTime =
    satelliteTemperature?.timestamp &&
    toRelativeTime(satelliteTemperature.timestamp);

  const degreeHeatingWeeks = degreeHeatingWeeksCalculator(
    degreeHeatingDays?.value
  );

  const metrics = [
    {
      label: "SURFACE TEMP",
      value: `${formatNumber(satelliteTemperature?.value, 1)}°C`,
    },
    {
      label: "HISTORICAL MAX",
      value: `${formatNumber(maxMonthlyMean, 1)}°C`,
      tooltipTitle: "Historical maximum monthly average over the past 20 years",
    },
    {
      label: "DEGREE HEATING WEEKS",
      value: formatNumber(degreeHeatingWeeks, 1),
      tooltipTitle:
        "Degree Heating Weeks - a measure of the amount of time above the 20 year historical maximum temperatures",
    },
    {
      label: "SST ANOMALY",
      tooltipTitle: "Difference between current SST and longterm average",
      value: `${
        sstAnomaly
          ? `${sstAnomaly > 0 ? "+" : ""}${formatNumber(sstAnomaly, 1)}`
          : "- -"
      }°C`,
    },
  ];

  return (
    <Card
      className={classes.card}
      style={{ backgroundColor: dhwColorFinder(degreeHeatingWeeks) }}
    >
      <CardHeader
        className={classes.header}
        title={
          <Grid container>
            <Grid item>
              <Typography className={classes.cardTitle} variant="h6">
                SATELLITE OBSERVATION
              </Typography>
            </Grid>
          </Grid>
        }
      />

      <CardContent className={classes.content}>
        <Box p="1rem" display="flex" flexGrow={1}>
          <Grid container spacing={1}>
            {metrics.map(({ label, value, tooltipTitle }) => (
              <Grid key={label} item xs={6}>
                <Typography
                  className={classes.contentTextTitles}
                  variant="subtitle2"
                >
                  {label}
                </Typography>
                <Tooltip title={tooltipTitle || ""}>
                  <Typography
                    className={classes.contentTextValues}
                    variant="h3"
                  >
                    {value}
                  </Typography>
                </Tooltip>
              </Grid>
            ))}
          </Grid>
        </Box>

        <Grid container>
          {dhwColorCode.map(({ value, color }) => (
            <Grid
              key={value}
              item
              xs={1}
              style={{ backgroundColor: `${color}`, height: "2rem" }}
            >
              <Box textAlign="center">
                <Typography variant="caption" align="center">
                  {value}
                </Typography>
              </Box>
            </Grid>
          ))}
        </Grid>

        <UpdateInfo
          relativeTime={relativeTime}
          timeText="Last data received"
          image={satellite}
          imageText="NOAA"
          live={false}
          frequency="daily"
          href="https://coralreefwatch.noaa.gov/"
        />
      </CardContent>
    </Card>
  );
};

const styles = () =>
  createStyles({
    ...incomingStyles,
    card: {
      ...incomingStyles.card,
      display: "flex",
      flexDirection: "column",
      height: "100%",
    },
    content: {
      display: "flex",
      flexDirection: "column",
      justifyContent: "space-between",
      flexGrow: 1,
      padding: 0,
    },
    contentText: {
      marginTop: "1rem",
      padding: "0 1rem",
    },
  });

interface SatelliteIncomingProps {
  maxMonthlyMean: number | null;
  liveData: LiveData;
}

type SatelliteProps = WithStyles<typeof styles> & SatelliteIncomingProps;

export default withStyles(styles)(Satellite);
