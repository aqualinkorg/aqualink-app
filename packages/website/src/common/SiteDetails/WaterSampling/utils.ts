import { formatNumber } from 'helpers/numberUtils';
import { Metrics, MetricsKeys, Sources } from 'store/Sites/types';
import { GridProps } from '@mui/material';
import siteServices from 'services/siteServices';
import { timeSeriesRequest } from 'store/Sites/helpers';
import { getSondeConfig } from 'constants/chartConfigs/sondeConfig';

type HUICardMetrics = Extract<
  Metrics,
  'salinity' | 'nitratePlusNitrite' | 'ph' | 'turbidity'
>;

export const watchColor = '#e5bb2bd0';
export const warningColor = '#ef883cd0';
export const alertColor = '#dd143ed0';

const thresholds = {
  nitratePlusNitrite: {
    good: 3.5,
    watch: 30,
    warning: 100,
  },
  turbidity: {
    good: 1,
    watch: 5,
    warning: 10,
  },
};

function getAlertColor(metric: HUICardMetrics, value?: number) {
  if (!value) return undefined;

  const compare = (th: { good: number; watch: number; warning: number }) => {
    if (value < th.good) return undefined;
    if (value < th.watch) return watchColor;
    if (value < th.warning) return warningColor;
    return alertColor;
  };

  switch (metric) {
    case 'nitratePlusNitrite':
      return compare(thresholds.nitratePlusNitrite);
    case 'turbidity':
      return compare(thresholds.turbidity);
    default:
      return undefined;
  }
}

function calculateGeometricMean(data: number[]): number | undefined {
  if (data.length === 0) return undefined;
  const lnSum = data.reduce((acc, curr) => acc + Math.log(curr), 0);
  return Math.exp(lnSum / data.length);
}

function calculateMean(data: number[]): number | undefined {
  if (data.length === 0) return undefined;
  const sum = data.reduce((acc, curr) => acc + curr);
  return sum / data.length;
}

export function getMeanCalculationFunction(
  source: Extract<Sources, 'hui' | 'sonde'>,
): (a: number[]) => number | undefined {
  switch (source) {
    case 'hui':
      return calculateGeometricMean;
    case 'sonde':
      return calculateMean;
    default:
      throw new Error(`Unknown source: ${source}`);
  }
}

const metricsForSource: Pick<
  { [Key in Sources]: MetricsKeys[] },
  'hui' | 'sonde'
> = {
  hui: ['turbidity', 'nitrate_plus_nitrite', 'ph', 'salinity'],
  sonde: [
    'odo_concentration',
    'cholorophyll_concentration',
    'ph',
    'salinity',
    'turbidity',
  ],
};

interface MetricField {
  label: string;
  value: string;
  unit?: string;
  color?: string;
  xs: GridProps['xs'];
}

export function metricFields(
  source: Extract<Sources, 'hui' | 'sonde'>,
  data?: Partial<Record<Metrics, number>>,
): MetricField[] {
  switch (source) {
    case 'hui':
      return [
        {
          label: 'Turbidity',
          value: `${formatNumber(data?.turbidity, 1)}`,
          unit: 'NTU',
          color: getAlertColor('turbidity', data?.turbidity),
          xs: 6,
        },
        {
          label: 'Nitrate Nitrite Nitrogen',
          value: `${formatNumber(data?.nitratePlusNitrite, 1)}`,
          unit: 'Âµg/L',
          color: getAlertColor('nitratePlusNitrite', data?.nitratePlusNitrite),
          xs: 6,
        },
        {
          label: 'pH',
          value: `${formatNumber(data?.ph, 1)}`,
          xs: 6,
        },
        {
          label: 'Salinity',
          value: `${formatNumber(data?.salinity, 1)}`,
          unit: source === 'hui' ? 'ppt' : 'psu',
          xs: 6,
        },
      ];

    case 'sonde':
      return [
        {
          label: 'DISSOLVED OXYGEN CONCENTRATION',
          value: formatNumber(data?.odoConcentration, 2),
          unit: getSondeConfig('odo_concentration').units,
          xs: 6,
        },
        {
          label: 'CHLOROPHYLL CONCENTRATION',
          value: formatNumber(data?.cholorophyllConcentration, 2),
          unit: getSondeConfig('cholorophyll_concentration').units,
          xs: 6,
        },
        {
          label: 'ACIDITY',
          value: formatNumber(data?.ph, 1),
          unit: getSondeConfig('ph').units,
          xs: 4,
        },
        {
          label: 'SALINITY',
          value: formatNumber(data?.salinity, 1),
          unit: getSondeConfig('salinity').units,
          xs: 5,
        },
        {
          label: 'TURBIDITY',
          value: formatNumber(data?.turbidity, 0),
          unit: getSondeConfig('turbidity').units,
          xs: 3,
        },
      ];
    default:
      throw new Error(`Unknown source: ${source}`);
  }
}

export async function getCardData(
  siteId: string,
  source: Extract<Sources, 'hui' | 'sonde'>,
) {
  try {
    const { data: uploadHistory } = await siteServices.getSiteUploadHistory(
      parseInt(siteId, 10),
    );

    const uploads =
      uploadHistory.filter(
        (x) =>
          x.dataUpload.sensorTypes.includes(source) ||
          // hui is specific type of sonde, look for hui as well when looking for sonde
          (source === 'sonde' && x.dataUpload.sensorTypes.includes('hui')),
      ) || [];
    if (uploads.length < 1) {
      return {};
    }

    switch (source) {
      case 'hui': {
        const now = new Date();
        const lastYear = now.setFullYear(now.getFullYear() - 1);
        const inLastYear = uploads.filter(
          ({ dataUpload: { maxDate } }) =>
            new Date(maxDate) > new Date(lastYear),
        );

        const minDate = inLastYear.reduce((min, curr) => {
          const currMin = curr.minDate || curr.dataUpload.minDate;
          return currMin < min ? currMin : min;
        }, new Date().toISOString());

        const maxDate =
          inLastYear.length > 0
            ? inLastYear.reduce((max, curr) => {
                const currMax = curr.maxDate || curr.dataUpload.maxDate;
                return currMax > max ? currMax : max;
              }, new Date(0).toISOString())
            : new Date().toISOString();

        const [data] = await timeSeriesRequest({
          siteId,
          start: minDate,
          end: maxDate,
          metrics: metricsForSource[source],
          hourly: true,
        });

        const pointId = inLastYear[0]?.surveyPoint;
        const samePoint =
          pointId !== null
            ? inLastYear.reduce(
                (acc, curr) => acc && curr.surveyPoint?.id === pointId.id,
                true,
              )
            : false;

        return {
          data,
          minDate,
          maxDate,
          point: samePoint ? pointId : undefined,
        };
      }
      case 'sonde': {
        // eslint-disable-next-line fp/no-mutating-methods
        uploads.sort((a, b) => {
          if (a.maxDate && b.maxDate) {
            const timeA = new Date(a.maxDate).getTime();
            const timeB = new Date(b.maxDate).getTime();
            return timeB - timeA;
          }
          return 0;
        });
        const { minDate, maxDate } = uploads[0].dataUpload;
        const { surveyPoint } = uploads[0];
        const [data] = await timeSeriesRequest({
          siteId,
          start: minDate,
          end: maxDate,
          metrics: metricsForSource[source],
          hourly: true,
        });
        return {
          data,
          minDate,
          maxDate,
          point: surveyPoint,
        };
      }
      default:
        throw new Error(`Unknown source: ${source}`);
    }
  } catch (err) {
    console.error(err);
    return {};
  }
}
