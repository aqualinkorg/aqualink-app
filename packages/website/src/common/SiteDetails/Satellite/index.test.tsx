import React from 'react';

import { parseLatestData } from 'store/Sites/helpers';
import { mockLatestData } from 'mocks/mockLatestData';
import { renderWithProviders } from 'utils/test-utils';
import Satellite from '.';

test('renders as expected', () => {
  const data = parseLatestData(mockLatestData);

  const { container } = renderWithProviders(
    <Satellite maxMonthlyMean={24} data={data} />,
  );
  expect(container).toMatchSnapshot();
});
