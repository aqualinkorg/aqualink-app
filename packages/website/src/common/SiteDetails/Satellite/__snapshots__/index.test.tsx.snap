// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <mock-card
    classname="Satellite-card-1"
    style="background-color: rgb(117, 180, 114);"
  >
    <div
      class="MuiCardHeader-root Satellite-header-3"
    >
      <div
        class="MuiCardHeader-content"
      >
        <mock-typography
          classname="MuiCardHeader-title"
          component="span"
          display="block"
          variant="h5"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <mock-typography
                classname="Satellite-cardTitle-2"
                variant="h6"
              >
                SATELLITE OBSERVATION
              </mock-typography>
            </div>
          </div>
        </mock-typography>
      </div>
    </div>
    <mock-cardcontent
      classname="Satellite-content-8"
    >
      <mock-box
        display="flex"
        flexgrow="1"
        p="1rem"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <mock-typography
              classname="Satellite-contentTextTitles-4"
              variant="subtitle2"
            >
              SURFACE TEMP
            </mock-typography>
            <mock-tooltip
              title=""
            >
              <mock-typography
                classname="Satellite-contentTextValues-5"
                variant="h3"
              >
                26.0°C
              </mock-typography>
            </mock-tooltip>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <mock-typography
              classname="Satellite-contentTextTitles-4"
              variant="subtitle2"
            >
              HISTORICAL MAX
            </mock-typography>
            <mock-tooltip
              title="Historical maximum monthly average over the past 20 years"
            >
              <mock-typography
                classname="Satellite-contentTextValues-5"
                variant="h3"
              >
                24.0°C
              </mock-typography>
            </mock-tooltip>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <mock-typography
              classname="Satellite-contentTextTitles-4"
              variant="subtitle2"
            >
              DEGREE HEATING WEEKS
            </mock-typography>
            <mock-tooltip
              title="Degree Heating Weeks - a measure of the amount of time above the 20 year historical maximum temperatures"
            >
              <mock-typography
                classname="Satellite-contentTextValues-5"
                variant="h3"
              >
                4.1
              </mock-typography>
            </mock-tooltip>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <mock-typography
              classname="Satellite-contentTextTitles-4"
              variant="subtitle2"
            >
              SST ANOMALY
            </mock-typography>
            <mock-tooltip
              title="Difference between current SST and longterm average"
            >
              <mock-typography
                classname="Satellite-contentTextValues-5"
                variant="h3"
              >
                - -°C
              </mock-typography>
            </mock-tooltip>
          </div>
        </div>
      </mock-box>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
          style="background-color: rgb(8, 91, 163); height: 2rem;"
        >
          <mock-box
            textalign="center"
          >
            <mock-typography
              align="center"
              variant="caption"
            >
              0
            </mock-typography>
          </mock-box>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
          style="background-color: rgb(12, 157, 165); height: 2rem;"
        >
          <mock-box
            textalign="center"
          >
            <mock-typography
              align="center"
              variant="caption"
            >
              2
            </mock-typography>
          </mock-box>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
          style="background-color: rgb(117, 180, 114); height: 2rem;"
        >
          <mock-box
            textalign="center"
          >
            <mock-typography
              align="center"
              variant="caption"
            >
              4
            </mock-typography>
          </mock-box>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
          style="background-color: rgb(123, 217, 37); height: 2rem;"
        >
          <mock-box
            textalign="center"
          >
            <mock-typography
              align="center"
              variant="caption"
            >
              6
            </mock-typography>
          </mock-box>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
          style="background-color: rgb(193, 229, 34); height: 2rem;"
        >
          <mock-box
            textalign="center"
          >
            <mock-typography
              align="center"
              variant="caption"
            >
              8
            </mock-typography>
          </mock-box>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
          style="background-color: rgb(229, 187, 43); height: 2rem;"
        >
          <mock-box
            textalign="center"
          >
            <mock-typography
              align="center"
              variant="caption"
            >
              10
            </mock-typography>
          </mock-box>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
          style="background-color: rgb(230, 121, 40); height: 2rem;"
        >
          <mock-box
            textalign="center"
          >
            <mock-typography
              align="center"
              variant="caption"
            >
              12
            </mock-typography>
          </mock-box>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
          style="background-color: rgb(232, 79, 46); height: 2rem;"
        >
          <mock-box
            textalign="center"
          >
            <mock-typography
              align="center"
              variant="caption"
            >
              14
            </mock-typography>
          </mock-box>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
          style="background-color: rgb(216, 69, 36); height: 2rem;"
        >
          <mock-box
            textalign="center"
          >
            <mock-typography
              align="center"
              variant="caption"
            >
              16
            </mock-typography>
          </mock-box>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
          style="background-color: rgb(128, 58, 32); height: 2rem;"
        >
          <mock-box
            textalign="center"
          >
            <mock-typography
              align="center"
              variant="caption"
            >
              18
            </mock-typography>
          </mock-box>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
          style="background-color: rgb(185, 41, 68); height: 2rem;"
        >
          <mock-box
            textalign="center"
          >
            <mock-typography
              align="center"
              variant="caption"
            >
              20
            </mock-typography>
          </mock-box>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
          style="background-color: rgb(128, 48, 126); height: 2rem;"
        >
          <mock-box
            textalign="center"
          >
            <mock-typography
              align="center"
              variant="caption"
            >
              25
            </mock-typography>
          </mock-box>
        </div>
      </div>
      <div
        class="MuiGrid-root UpdateInfo-updateInfo-10 false MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root UpdateInfo-updateIcon-12 MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 10.12h-6.78l2.74-2.82c-2.73-2.7-7.15-2.8-9.88-.1-2.73 2.71-2.73 7.08 0 9.79s7.15 2.71 9.88 0C18.32 15.65 19 14.08 19 12.1h2c0 1.98-.88 4.55-2.64 6.29-3.51 3.48-9.21 3.48-12.72 0-3.5-3.47-3.53-9.11-.02-12.58s9.14-3.47 12.65 0L21 3v7.12zM12.5 8v4.25l3.5 2.08-.72 1.21L11 13V8h1.5z"
                />
              </svg>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <mock-box
                display="flex"
                flexdirection="column"
              >
                <mock-typography
                  classname="UpdateInfo-updateInfoText-13"
                  variant="caption"
                >
                  Last data received
                   
                  5 min. ago
                </mock-typography>
                <mock-typography
                  classname="UpdateInfo-updateInfoText-13"
                  variant="caption"
                >
                  Updated 
                  daily
                </mock-typography>
              </mock-box>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root UpdateInfo-nooaChip-14 MuiGrid-item"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
          >
            <a
              class="UpdateInfo-nooaLink-15"
              href="https://coralreefwatch.noaa.gov/"
              target="_blank"
            >
              <mock-typography
                classname="UpdateInfo-nooaChipText-16"
              >
                NOAA
              </mock-typography>
              <img
                alt="sensor-type"
                class="UpdateInfo-sensorImage-17"
                src="satellite.svg"
              />
            </a>
          </div>
        </div>
      </div>
    </mock-cardcontent>
  </mock-card>
</div>
`;
